lib/failure.sh

```diff
@@ -361,6 +361,77 @@ fail_invalid_semver() {
   fi
 }
 
+# Yarn 2 failures
+
+fail_missing_yarnrc_yml() {
+  local build_dir="$1"
+
+  if [[ ! -f "$build_dir/.yarnrc.yml" ]]; then
+    mcount "failures.missing-yarnrc-yml"
+    meta_set "failure" "missing-yarnrc-yml"
+    header "Build failed"
+    warn "The yarnrc.yml file could not found
+
+      It looks like the yarnrc.yml file is missing from this project. Please
+      make sure this file is checked into version control and made available to
+      Heroku.
+
+      To generate yarnrc.yml, make sure Yarn 2 is installed on your local
+      machine and set the version in your project directory with:
+
+       $ yarn set version berry
+      "
+    fail
+  fi
+}
+
+fail_missing_yarn_path() {
+  local build_dir="$1"
+  local yarn_path="$2"
+
+  if [[ "$yarn_path" == "" ]]; then
+    mcount "failures.missing-yarn-path"
+    meta_set "failure" "missing-yarn-path"
+    header "Build failed"
+    warn "The 'yarnPath' could not be read from the yarnrc.yml file
+
+      It looks like yarnrc.yml is missing the 'yarnPath' value, which is needed
+      to identify the location of yarn for this build.
+
+      To generate properly set 'yarnPath', make sure Yarn 2 is installed on your
```

Comment:
"To generate properly set 'yarnPath'..." reads oddly to me, but maybe that because I don't know what yarnPath is.